generated/nixnetsocket/nixnetsocket.proto

```diff
@@ -21,15 +21,29 @@ service NiXnetSocket {
   rpc Close(CloseRequest) returns (CloseResponse);
   rpc GetLastErrorNum(GetLastErrorNumRequest) returns (GetLastErrorNumResponse);
   rpc GetLastErrorStr(GetLastErrorStrRequest) returns (GetLastErrorStrResponse);
+  rpc GetSocketOption(GetSocketOptionRequest) returns (GetSocketOptionResponse);
   rpc IsSet(IsSetRequest) returns (IsSetResponse);
   rpc Select(SelectRequest) returns (SelectResponse);
+  rpc SetSocketOption(SetSocketOptionRequest) returns (SetSocketOptionResponse);
   rpc Socket(SocketRequest) returns (SocketResponse);
 }
 
 enum NiXnetSocketAttribute {
   NIXNETSOCKET_ATTRIBUTE_UNSPECIFIED = 0;
 }
 
+enum SocketOptions {
```

Comment:
TODO: This isn't the exhaustive list of options that can be used in Get or SetSocketOption. It seems like it can at least also be used for some TCP options like TCP_NO_DELAY (see [here in SocketTests.cpp](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/xnet/xnetCored/nixntipstack_tests/source/nixntipstackt/SocketTests.cpp&version=GC72eb6a7f4204d6ae570dfcd207c804c1d7817ca5&line=3665&lineEnd=3666&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)).

Reply:
OK, that is at least nxTCP_NODELAY. I only see one such option in the header. It we *can't* get an exhaustive list: that's a problem for this implementation.